lib/datasource/docker/index.ts

```diff
@@ -535,3 +543,35 @@ export async function getPkgReleases({
   }
   return ret;
 }
+
+function getECRAuthToken(region: string, opts: hostRules.HostRule) {
+  const config = { region, accessKeyId: undefined, secretAccessKey: undefined };
+  if (opts.username && opts.password) {
+    config.accessKeyId = opts.username;
+    config.secretAccessKey = opts.password;
+  }
+  const ecr = new AWS.ECR(config);
+  return new Promise<string>((resolve, reject) => {
+    ecr.getAuthorizationToken({}, (err, data) => {
+      if (err) {
+        logger.error({ err });
+        resolve(null);
+      }
+      logger.debug({ data }, 'getAuthorizationToken response');
+      const authorizationToken =
+        data &&
+        data.authorizationData &&
+        data.authorizationData[0] &&
+        data.authorizationData[0].authorizationToken;
+      if (authorizationToken) {
+        resolve(authorizationToken);
+      } else {
+        logger.error(
```

Comment:
If credentials are incorrect, what do we expect ECR to return? e.g. `err`, or empty `data`? Should data contain an authorization token 100% of the time it's returned?

Reply:
Yes, `err` for incorrect credentials. 

> Should data contain an authorization token 100% of the time it's returned?

That's my understanding. Incorrect credentials & insufficient permission should result in an error. There will only be one `authorizationData` element as it's using the default registry from the supplied region. See https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/ECR.html#getAuthorizationToken-property